ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/IcebergCompactionExecutor.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.hadoop.hive.ql.txn.compactor;
+
+import org.apache.hadoop.hive.conf.HiveConf;
+import org.apache.hadoop.hive.metastore.IMetaStoreClient;
+import org.apache.hadoop.hive.metastore.api.StorageDescriptor;
+import org.apache.hadoop.hive.metastore.api.Table;
+import org.apache.hadoop.hive.metastore.txn.CompactionInfo;
+import org.apache.hadoop.hive.metastore.txn.TxnUtils;
+import org.apache.thrift.TException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+
+public class IcebergCompactionExecutor implements CompactionExecutor {
+  static final private String CLASS_NAME = Worker.class.getName();
+  static final private Logger LOG = LoggerFactory.getLogger(CLASS_NAME);
+
+  private final Worker worker;
+  private final IMetaStoreClient msc;
+  private final HiveConf conf;
+  private final CompactionInfo ci;
+  private final Table table;
+  private final boolean collectGenericStats;
+  private final boolean collectMrStats;
+  private final Worker.CompactionTxn compactionTxn;
+
+  public IcebergCompactionExecutor(Worker worker, CompactionInfo ci, Worker.CompactionTxn compactionTxn, 
```

Comment:
there is no compactionTxn &  collectMrStats  for Iceberg

Reply:
boolean genericStats = conf.getBoolVar(HiveConf.ConfVars.HIVE_COMPACTOR_GATHER_STATS);
    boolean mrStats = conf.getBoolVar(HiveConf.ConfVars.HIVE_MR_COMPACTOR_GATHER_STATS);

Reply:
genericStats and mrStats values are decided based on HiveConf values in the Worker, why is mrStats not relevant for Iceberg?

```
    boolean genericStats = conf.getBoolVar(HiveConf.ConfVars.HIVE_COMPACTOR_GATHER_STATS);
    boolean mrStats = conf.getBoolVar(HiveConf.ConfVars.HIVE_MR_COMPACTOR_GATHER_STATS);
```

Reply:
`mr` means - MapReduce, that mode won't be supported for Iceberg tables

Reply:
Done